applications/fishing-map/src/features/map/Popup.tsx

```diff
@@ -23,20 +23,25 @@ function PopupWrapper({
   onClose,
   loading = false,
 }: PopupWrapper) {
-  const { updateDataviewConfig } = useDataviewsConfigConnect()
-  const onVesselClick = (vessel: any) => {
-    const dataviewConfig: WorkspaceDataviewConfig = {
-      id: `track-${vessel.id}`,
-      dataviewId: 2,
-      datasetsConfig: [
-        {
-          datasetId: TRACKS_DATASET_ID,
-          params: [{ id: 'vesselId', value: vessel.id }],
-          endpoint: 'carriers-tracks',
-        },
-      ],
+  const { upsertDataviewInstance } = useDataviewInstancesConnect()
+  const onVesselClick = (vessel: any, feature: TooltipEventFeature) => {
+    const datasetId = feature.dataset?.relatedDatasets?.find(
+      (relatedDataset) => relatedDataset.type === TRACKS_DATASET_TYPE
+    )?.id
+    if (datasetId) {
+      const dataviewInstance: DataviewInstance = {
+        id: `track-${vessel.id}`,
```

Comment:
The logic to generate this should be centralized somewhere, duplicate with Search.tsx 
https://github.com/GlobalFishingWatch/frontend/pull/124/files#diff-0fc787fc5bdbf9885682179d3112abad8749adc32b96fd9b5f511b878b7c089eR70

Reply:
It isn't easy as we have to configure the dataset for the track and the same for the info with different endpoints for each one... I've created a `getVesselDataviewInstance` util in dataviews to centralize but will need some review later